src/ast.hh

```diff
@@ -294,6 +296,8 @@ class AST {
   Product::Base * get_backtrack_product() {
       return backtrack_product;
   }
+
+  Printer::Checkpoint *checkpoint;
```

Comment:
The `Checkpoint` class is a member of the `AST` class, because the `AST` can be accessed from all of the printer functions that write the code to the `.hh` file and I need a way/access point to call the methods defined in the `Checkpoint` class.
I guess I could also refactor and store a ptr to the `Checkpoint` class in the `Printer::Cpp` class and call everything from there without accessing the `AST`. Your call.